pkgs/development/python-modules/agate-sql/default.nix

```
@@ -1,15 +1,30 @@
-{ lib, fetchPypi, buildPythonPackage, agate, sqlalchemy, crate }:
+{ lib
+, buildPythonPackage
+, isPy27
+, fetchFromGitHub
+, agate
+, sqlalchemy
+, crate
+, nose
+, geojson
+}:
 
 buildPythonPackage rec {
   pname = "agate-sql";
-  version = "0.5.5";
+  version = "0.5.6";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "50a39754babef6cd0d1b1e75763324a49593394fe46ab1ea9546791b5e6b69a7";
+  disabled = isPy27;
+
+  src = fetchFromGitHub {
+    owner = "wireservice";
+    repo = "agate-sql";
+    rev = version;
+    sha256 = "16rijcnvxrvw9mmyk4228dalrr2qb74y649g1l6qifiabx5ij78s";
   };
 
-  propagatedBuildInputs = [ agate sqlalchemy crate ];
+  propagatedBuildInputs = [ agate sqlalchemy ];
+
+  checkInputs = [ crate nose geojson ];
```

```suggestion
  checkInputs = [ crate nose geojson ];
  
  checkPhase = ''
  	nosetests
  '';
```

I am not sure how the tests are executed with setuptools.

---

I'm not sure either, but I was okay with the fact that it just worked.

---

setuptools tests (`python setup.py tests`) are not executed anymore because setuptools no longer supports this. An external test runner is always needed nowadays.